### ISSUE TITLE
训练卡住不动，等了几分钟也没反应。

### ISSUE THREAD

**Issue by leon2235** [modelscope/ms-swift#6984]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)

运行的sh:

CUDA_VISIBLE_DEVICES=0,1,2,3 \
NPROC_PER_NODE=4 \
swift rlhf \
    --rlhf_type grpo \
    --model '/public/home/202420164005/model/Qwen/Qwen2.5-Omni-7B' \
    --adapters '/public/home/202420164005/code/ms-swift/SFT/checkpoints/v3-20251203-105243/checkpoint-180' \
    --external_plugins '/public/home/202420164005/code/ms-swift/GRPO/reward.py' \
    --reward_funcs external_ad_content_llm_remote \
    --train_type lora \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --torch_dtype bfloat16 \
    --dataset '/public/home/202420164005/code/ms-swift/GRPO/data/train_gemini_rewrite.jsonl' \
    --val_dataset '/public/home/202420164005/code/ms-swift/GRPO/data/val_gemini_rewrite.jsonl' \
    --max_length 4096 \
    --max_completion_length 800 \
    --num_train_epochs 5 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --learning_rate 1e-5 \
    --gradient_accumulation_steps 1 \
    --eval_steps 20 \
    --save_steps 20 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --output_dir '/public/home/202420164005/code/ms-swift/GRPO/checkpoint_1128' \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --num_generations 4 \
    --temperature 0.9 \
    --system '你是一名顶尖的语音语言学和认知神经科学专家。' \
    --log_completions true

终端输出：

run sh: `/public/home/202420164005/miniconda3/envs/swift-env/bin/python3.10 -m torch.distributed.run --nproc_per_node 4 /public/home/202420164005/code/ms-swift/swift/cli/rlhf.py --rlhf_type grpo --model /public/home/202420164005/model/Qwen/Qwen2.5-Omni-7B --adapters /public/home/202420164005/code/ms-swift/SFT/checkpoints/v3-20251203-105243/checkpoint-180 --external_plugins /public/home/202420164005/code/ms-swift/GRPO/reward.py --reward_funcs external_ad_content_llm_remote --train_type lora --lora_rank 8 --lora_alpha 32 --target_modules all-linear --torch_dtype bfloat16 --dataset /public/home/202420164005/code/ms-swift/GRPO/data/train_gemini_rewrite.jsonl --val_dataset /public/home/202420164005/code/ms-swift/GRPO/data/val_gemini_rewrite.jsonl --max_length 4096 --max_completion_length 800 --num_train_epochs 5 --per_device_train_batch_size 4 --per_device_eval_batch_size 4 --learning_rate 1e-5 --gradient_accumulation_steps 1 --eval_steps 20 --save_steps 20 --save_total_limit 2 --logging_steps 5 --output_dir /public/home/202420164005/code/ms-swift/GRPO/checkpoint_1128 --warmup_ratio 0.05 --dataloader_num_workers 4 --num_generations 4 --temperature 0.9 --system 你是一名顶尖的语音语言学和认知神经科学专家。 --log_completions true`
[INFO:swift] Successfully registered `/public/home/202420164005/code/ms-swift/swift/llm/dataset/data/dataset_info.json`.
[INFO:swift] Setting args.remove_unused_columns: False
[INFO] Using scorer model: gemini-2.5-flash
[INFO] API endpoint: https://api.apiyi.com/v1/chat/completions
[INFO] Using scorer model: gemini-2.5-flash
[INFO] API endpoint: https://api.apiyi.com/v1/chat/completions
[INFO] Using scorer model: gemini-2.5-flash
[INFO] API endpoint: https://api.apiyi.com/v1/chat/completions
[INFO] Using scorer model: gemini-2.5-flash
[INFO] API endpoint: https://api.apiyi.com/v1/chat/completions
[INFO:swift] Successfully imported external_plugins: ['/public/home/202420164005/code/ms-swift/GRPO/reward.py'].
[INFO:swift] rank: 0, local_rank: 0, world_size: 4, local_world_size: 4
Unrecognized keys in `rope_scaling` for 'rope_type'='default': {'mrope_section'}
Unrecognized keys in `rope_scaling` for 'rope_type'='default': {'mrope_section'}
Unrecognized keys in `rope_scaling` for 'rope_type'='default': {'mrope_section'}
Unrecognized keys in `rope_scaling` for 'rope_type'='default': {'mrope_section'}
`torch_dtype` is deprecated! Use `dtype` instead!
`torch_dtype` is deprecated! Use `dtype` instead!
`torch_dtype` is deprecated! Use `dtype` instead!
`torch_dtype` is deprecated! Use `dtype` instead!
[INFO:swift] Because len(args.val_dataset) > 0, setting split_dataset_ratio: 0.0
[INFO:swift] Setting args.lazy_tokenize: True

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by hjh0119** [comment]
run normally on my end, plz provide pip list 
```
CUDA_VISIBLE_DEVICES=0,1,2,3 \
NPROC_PER_NODE=4 \
swift rlhf \
    --rlhf_type grpo \
    --model 'Qwen/Qwen2.5-Omni-7B' \
    --reward_funcs format \
    --train_type lora \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --torch_dtype bfloat16 \
    --dataset 'AI-ModelScope/clevr_cogen_a_train#1000' \
    --max_length 4096 \
    --max_completion_length 800 \
    --num_train_epochs 5 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --learning_rate 1e-5 \
    --gradient_accumulation_steps 1 \
    --eval_steps 20 \
    --save_steps 20 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --num_generations 4 \
    --temperature 0.9 \
    --system '你是一名顶尖的语音语言学和认知神经科学专家。' \
    --log_completions true
```

**Comment by leon2235** [comment]
> run normally on my end, plz provide pip list
> 
> ```
> CUDA_VISIBLE_DEVICES=0,1,2,3 \
> NPROC_PER_NODE=4 \
> swift rlhf \
>     --rlhf_type grpo \
>     --model 'Qwen/Qwen2.5-Omni-7B' \
>     --reward_funcs format \
>     --train_type lora \
>     --lora_rank 8 \
>     --lora_alpha 32 \
>     --target_modules all-linear \
>     --torch_dtype bfloat16 \
>     --dataset 'AI-ModelScope/clevr_cogen_a_train#1000' \
>     --max_length 4096 \
>     --max_completion_length 800 \
>     --num_train_epochs 5 \
>     --per_device_train_batch_size 4 \
>     --per_device_eval_batch_size 4 \
>     --learning_rate 1e-5 \
>     --gradient_accumulation_steps 1 \
>     --eval_steps 20 \
>     --save_steps 20 \
>     --save_total_limit 2 \
>     --logging_steps 5 \
>     --warmup_ratio 0.05 \
>     --dataloader_num_workers 4 \
>     --num_generations 4 \
>     --temperature 0.9 \
>     --system '你是一名顶尖的语音语言学和认知神经科学专家。' \
>     --log_completions true
> ```


my pip list:
(swift-env) [202420164005@admin model]$ pip list
Package                           Version
--------------------------------- -------------
absl-py                           2.3.1
accelerate                        1.11.0
addict                            2.4.0
aiofiles                          24.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.2
aiosignal                         1.4.0
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-doc                     0.0.3
annotated-types                   0.7.0
antlr4-python3-runtime            4.13.2
anyio                             4.11.0
astor                             0.8.1
async-timeout                     5.0.1
attrdict                          2.0.1
attrs                             25.4.0
audioread                         3.1.0
av                                16.0.1
binpacking                        1.5.2
blake3                            1.0.8
Brotli                            1.1.0
cachetools                        6.2.1
cbor2                             5.7.1
certifi                           2025.10.5
cffi                              2.0.0
charset-normalizer                3.4.4
click                             8.2.1
cloudpickle                       3.1.2
compressed-tensors                0.11.0
contourpy                         1.3.2
cpm-kernels                       1.0.11
crcmod                            1.7
cryptography                      46.0.3
cupy-cuda12x                      13.6.0
cycler                            0.12.1
dacite                            1.9.2
datasets                          3.6.0
decorator                         5.2.1
decord                            0.6.0
depyf                             0.19.0
dill                              0.3.8
diskcache                         5.6.3
distro                            1.9.0
dnspython                         2.8.0
einops                            0.8.1
email-validator                   2.3.0
exceptiongroup                    1.3.0
fastapi                           0.121.0
fastapi-cli                       0.0.14
fastapi-cloud-cli                 0.3.1
fastrlock                         0.8.3
ffmpy                             0.6.4
filelock                          3.20.0
fonttools                         4.60.1
frozendict                        2.4.6
frozenlist                        1.8.0
fsspec                            2025.3.0
future                            1.0.0
gguf                              0.17.1
gradio                            5.49.1
gradio_client                     1.13.3
groovy                            0.1.2
grpcio                            1.76.0
h11                               0.16.0
hf-xet                            1.2.0
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
huggingface-hub                   0.36.0
idna                              3.11
importlib_metadata                8.7.0
interegular                       0.3.3
jieba                             0.42.1
Jinja2                            3.1.6
jiter                             0.11.1
jmespath                          0.10.0
joblib                            1.5.2
json_repair                       0.52.4
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
kiwisolver                        1.4.9
lark                              1.2.2
latex2sympy2_extended             1.0.6
lazy_loader                       0.4
librosa                           0.11.0
llguidance                        0.7.30
llvmlite                          0.44.0
lm-format-enforcer                0.11.3
Markdown                          3.9
markdown-it-py                    4.0.0
MarkupSafe                        3.0.3
math-verify                       0.5.2
matplotlib                        3.10.7
mdurl                             0.1.2
mistral_common                    1.8.5
modelscope                        1.32.0
mpmath                            1.3.0
ms_swift                          3.10.0.dev0
msgpack                           1.1.2
msgspec                           0.19.0
multidict                         6.7.0
multiprocess                      0.70.16
mutagen                           1.47.0
networkx                          3.4.2
ninja                             1.13.0
nltk                              3.9.2
numba                             0.61.2
numpy                             2.2.6
nvidia-cublas-cu12                12.8.4.1
nvidia-cuda-cupti-cu12            12.8.90
nvidia-cuda-nvrtc-cu12            12.8.93
nvidia-cuda-runtime-cu12          12.8.90
nvidia-cudnn-cu12                 9.10.2.21
nvidia-cufft-cu12                 11.3.3.83
nvidia-cufile-cu12                1.13.1.3
nvidia-curand-cu12                10.3.9.90
nvidia-cusolver-cu12              11.7.3.90
nvidia-cusparse-cu12              12.5.8.93
nvidia-cusparselt-cu12            0.7.1
nvidia-ml-py                      13.580.82
nvidia-nccl-cu12                  2.27.3
nvidia-nvjitlink-cu12             12.8.93
nvidia-nvshmem-cu12               3.3.20
nvidia-nvtx-cu12                  12.8.90
nvitop                            1.5.3
omegaconf                         2.3.0
openai                            2.6.1
openai-harmony                    0.0.8
opencv-python-headless            4.12.0.88
orjson                            3.11.4
oss2                              2.19.1
outlines_core                     0.2.11
packaging                         25.0
pandas                            2.3.3
partial-json-parser               0.2.1.1.post6
peft                              0.17.1
pillow                            11.3.0
pip                               25.2
platformdirs                      4.5.0
pooch                             1.8.2
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.4.1
protobuf                          6.33.0
psutil                            7.1.3
py-cpuinfo                        9.0.0
pyarrow                           22.0.0
pybase64                          1.4.2
pycountry                         24.6.1
pycparser                         2.23
pycryptodome                      3.23.0
pydantic                          2.11.10
pydantic_core                     2.33.2
pydantic-extra-types              2.10.6
pydub                             0.25.1
Pygments                          2.19.2
pyparsing                         3.2.5
python-dateutil                   2.9.0.post0
python-dotenv                     1.2.1
python-json-logger                4.0.0
python-multipart                  0.0.20
pytz                              2025.2
PyYAML                            6.0.3
pyzmq                             27.1.0
qwen-omni-utils                   0.0.8
ray                               2.51.1
referencing                       0.37.0
regex                             2025.10.23
requests                          2.32.5
rich                              14.2.0
rich-toolkit                      0.15.1
rignore                           0.7.6
rouge                             1.0.1
rpds-py                           0.28.0
ruff                              0.14.3
safehttpx                         0.1.7
safetensors                       0.6.2
scikit-learn                      1.7.2
scipy                             1.15.3
semantic-version                  2.10.0
sentencepiece                     0.2.1
sentry-sdk                        2.43.0
setproctitle                      1.3.7
setuptools                        80.9.0
shellingham                       1.5.4
simplejson                        3.20.2
six                               1.17.0
sniffio                           1.3.1
sortedcontainers                  2.4.0
soundfile                         0.13.1
soxr                              1.0.0
starlette                         0.49.3
sympy                             1.14.0
tensorboard                       2.20.0
tensorboard-data-server           0.7.2
threadpoolctl                     3.6.0
tiktoken                          0.12.0
tokenizers                        0.22.1
tomlkit                           0.13.3
torch                             2.8.0
torchaudio                        2.8.0
torchvision                       0.23.0
tqdm                              4.67.1
transformers                      4.57.1
transformers-stream-generator     0.0.5
triton                            3.4.0
trl                               0.24.0
typer                             0.20.0
typing_extensions                 4.15.0
typing-inspection                 0.4.2
tzdata                            2025.2
urllib3                           2.5.0
uvicorn                           0.38.0
uvloop                            0.22.1
vllm                              0.11.0
watchfiles                        1.1.1
websockets                        15.0.1
Werkzeug                          3.1.3
wheel                             0.45.1
xformers                          0.0.32.post1
xgrammar                          0.1.25
xxhash                            3.6.0
yarl                              1.22.0
zipp                              3.23.0
zstandard                         0.25.0

**Comment by hjh0119** [comment]
pip install -U ms-swift

**Comment by leon2235** [comment]
> pip install -U ms-swift

刚刚试了一下单卡，是没问题的，我在找找原因

**Comment by robotsp** [comment]
请问找到原因了吗， @leon2235 我这边训了几步就会卡住
